### DISCUSSION TITLE
[Voice] How to get the number of users in a voice channel? v3

### DISCUSSION THREAD

**Discussion by lifeisperfecto** [diamondburned/arikawa#314] [Help] [answered]
Hi, using the v3 api I can use `state.Channel(id discord.ChannelD)` to get a `*discord.Channel` object, but that only returns the number of users in a thread which only works if the channel is a thread. 

Is it possible to get the number of users in a voice channel based on the channel ID?

The problem I'm trying to solve is that, if it's been 5 minutes since there's been zero users in the voice channel then I want to disconnect due to inactivity.

Thanks

**Comment by diamondburned** [comment, marked as answer]
That would be [`state.VoiceStates(discord.GuildID)`](https://pkg.go.dev/github.com/diamondburned/arikawa/v3@v3.0.0-rc.5/state/store#VoiceStateStore), which would give you a slice of `discord.VoiceState` instance that you can use to check if anyone's in any voice channel.

Note that `VoiceStates` is part of `store.VoiceStateStore`:

```
- state.State
  - session.Session
    - api.Client
    - gateway.Gateway
  - store.Cabinet
    - store.VoiceStateStore <--

**Reply by lifeisperfecto** [reply]
Thanks
